apps/src/lib/node/ledger/shell/vote_extensions.rs

```diff
@@ -8,6 +8,11 @@ mod extend_votes {
 
     type SignedExt = Signed<VoteExtension>;
 
+    pub enum ValidateVoteExtensionResult {
+        Success,
+        Failure { reason: String },
+    }
```

Comment:
This is crude but thinking something like this we can log (maybe a better variant for each possible reason?). I thought maybe `Result<()>` and logging the `Err(_)` but if `validate_vote_extension` is passed an invalid vote extension and we verify it as not valid, that isn't an error case but an expected execution path.